File: My_solutions/00_incredible_chairs.py
import pyomo.environ as pyo

# Create a model
model = pyo.ConcreteModel()

# Define variables using a list of indices
chairs = ['xA', 'xB']
model.x = pyo.Var(chairs, domain=pyo.NonNegativeReals)

# Define objective: Maximize 4*x['xA'] + 6*x['xB']
model.obj = pyo.Objective(expr=4 * model.x['xA'] + 6 * model.x['xB'], sense=pyo.maximize)

# Define constraints
model.c1 = pyo.Constraint(expr=2 * model.x['xA'] <= 14)
model.c2 = pyo.Constraint(expr=3 * model.x['xB'] <= 15)
model.c3 = pyo.Constraint(expr=4 * model.x['xA'] + 3 * model.x['xB'] <= 36)

# Print the model (similar to print(IC) in Julia)
model.pprint()

# Solve the model
# Note: You need a solver like 'highs', 'glpk', or 'gurobi' installed on your system
solver = pyo.SolverFactory('gurobi') 
results = solver.solve(model, tee=True)

# Print results
print(f"Termination status: {results.solver.termination_condition}")

if results.solver.termination_condition == pyo.TerminationCondition.optimal:
    print(f"Optimal objective value: {pyo.value(model.obj)}")
    for p in chairs:
        print(f"{p}: {pyo.value(model.x[p])}")
else:
    print("No optimal solution available")

File: My_solutions/00_incredible_chairs.jl
using JuMP
using HiGHS

chairs = [:xA, :xB]

IC = Model(HiGHS.Optimizer)
set_optimizer_attribute(IC, "output_flag", true)  # solver log (similar to tee=True)

@variable(IC, x[chairs] >= 0)

@objective(IC, Max, 4 * x[:xA] + 6 * x[:xB])

@constraint(IC, c1, 2 * x[:xA] <= 14)
@constraint(IC, c2, 3 * x[:xB] <= 15)
@constraint(IC, c3, 4 * x[:xA] + 3 * x[:xB] <= 36)

print(IC)
println()
optimize!(IC)

println("Termination status: $(termination_status(IC))")

if is_solved_and_feasible(IC)
    println("Optimal objective value: $(objective_value(IC))")
    for p in chairs
        println("$(p): $(value(x[p]))")
    end
else
    println("No optimal solution available")
end
